### ISSUE TITLE
Components with complex selectors report unrecognized tagname when used

### ISSUE THREAD

**Issue by zoechi** [dart-lang/angular_analyzer_plugin#171]
The component is listed in `directives: [BinaryResourceComponent]`

```dart
@Component(
    selector: 'binary-resource,[binary-resource]',
```

**Comment by MichaelRFairhurst** [comment]
Ah, this is an interesting bug.

In our code we check each directive that matches the element, and if `directive.selector is TagNameSelector` then we call the tag good. This prevents the case of someone using, say, `<asdf *ngIf="...` and marking that as a resolved tag when it isn't.

However, in this case (and I'm sure this will come up more than just here), `directive.selector` is NOT `TagNameSelector`, its `OrSelector`.  So we don't respect that the tag name was matched.

This is a bit tricky, we can't just check that the selector AST _contains_ `TagNameSelector` either, or (in this case) `<asdk binary-resource` would also fail to detect the unresolved tag. Rather, `selector.match` needs to track if it matched based on the tag name, so it can't simply return 'bool', but rather probably something like a new object. 

Note we need to handle the edge case of where `asdf[prop],[otherprop]` matches both `asdf` and `otherprop`, but not `prop` and therefore the fact that `asdf` was matched in a directive needs to be ignored...though that's very edge case.

**Comment by MichaelRFairhurst** [comment]
Ping @mk13, do you think you could solve this after #140 ?

**Comment by MichaelRFairhurst** [comment]
I'm gonna take this one on

**Comment by MichaelRFairhurst** [comment]
Merged #236

**Comment by zoechi** [comment]
Is there anything work in progress for being able to use the plugin with newer Dart version?
I saw it mentioned that the analyzyer plugin architecture will take a few months to become available.
Is this what we need to wait for?

**Comment by MichaelRFairhurst** [comment]
Nothing available yet, but it shouldn't be multiple months away.

Regarding if the plugin arch is the problem, bit of yes and a bit of no. Basically task scheduling and caching were changed, making the old plugin system incompatible. However, even with a new plugin system, we still have to adapt to the new task scheduling and caching changes. At this point, we've actually done the latter mostly but not much on the former...at this point we may create a temporary architecture so that its usable quickly and we can develop the real plugin arch with a bit more care.

**Comment by zoechi** [comment]
Sounds great! Thanks a lot for the feedback. Very helpful for planning.
